doc/source/serve/doc_code/hpu_inference_serve_deepspeed.py

```diff
@@ -0,0 +1,272 @@
+        write_checkpoints_json(
+            self.model_id_or_path, self._local_rank, checkpoints_json, token=""
+        )
+
+        # Prepare DeepSpeed inference configuration
+        kwargs = {"dtype": torch.bfloat16}
+        kwargs["checkpoint"] = checkpoints_json.name
+        kwargs["tensor_parallel"] = {"tp_size": self._world_size}
+        # Enable hpu graph, similar to cuda graph
+        kwargs["enable_cuda_graph"] = True
+        # Specify injection policy, required by DeepSpeed tensor parallelism
+        kwargs["injection_policy"] = get_ds_injection_policy(self.model_config)
+
+        # Initialize the inference engine
+        self.model = deepspeed.init_inference(model, **kwargs).module
+
+    def tokenize(self, prompt: str):
+        """Tokenize the input and move it to HPU."""
+
+        input_tokens = self.tokenizer(prompt, return_tensors="pt", padding=True)
+        return input_tokens.input_ids.to(device=self.device)
+
+    def generate(self, prompt: str, **config: Dict[str, Any]):
+        """Take in a prompt and generate a response."""
+
+        input_ids = self.tokenize(prompt)
+        gen_tokens = self.model.generate(input_ids, **config)
+        return self.tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)[0]
+
+    def streaming_generate(self, prompt: str, streamer, **config: Dict[str, Any]):
+        """Generate streamed response given an input."""
+
+        input_ids = self.tokenize(prompt)
+        self.model.generate(input_ids, streamer=streamer, **config)
+
+    def get_streamer(self):
+        """Return a streamer.
+
+        Only the rank 0 worker's result is needed.
+        Other workers return a fake streamer.
+        """
+
+        if self._local_rank == 0:
+            return RayTextIteratorStreamer(self.tokenizer, skip_special_tokens=True)
+        else:
+
+            class FakeStreamer:
+                def put(self, value):
+                    pass
+
+                def end(self):
+                    pass
+
+            return FakeStreamer()
+
+
+class RayTextIteratorStreamer(TextStreamer):
+    def __init__(
+        self,
+        tokenizer,
+        skip_prompt: bool = False,
+        timeout: int = None,
+        **decode_kwargs: Dict[str, Any],
+    ):
+        super().__init__(tokenizer, skip_prompt, **decode_kwargs)
+        self.text_queue = Queue()
+        self.stop_signal = None
+        self.timeout = timeout
+
+    def on_finalized_text(self, text: str, stream_end: bool = False):
+        self.text_queue.put(text, timeout=self.timeout)
+        if stream_end:
+            self.text_queue.put(self.stop_signal, timeout=self.timeout)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        value = self.text_queue.get(timeout=self.timeout)
+        if value == self.stop_signal:
+            raise StopIteration()
+        else:
+            return value
+
+
+# __worker_def_end__
+
+# __deploy_def_start__
+# Required variables for DeepSpeed on HPU
```

Comment:
```suggestion
# Required variables for DeepSpeed on HPU.
```

Reply:
Who requires the variables? If you can remove the passive voice, you could be more precise.